code/Domain/Services/SchedulingService.cs

```diff
@@ -0,0 +1,247 @@
+        {
+            return;
+        }
+
+        string jobId = job.Id.ToString();
+        JobKey jobKey = new(jobId);
+
+        if (await _scheduler.CheckExists(jobKey))
+        {
+            _logger.LogCritical(
+                "Found existing job with id {Id} while scheduling. Scheduling aborted",
+                jobId);
+            return;
+        }
+
+        _logger.LogInformation("Scheduling job {JobId}", jobId);
+
+        IJobDetail jobDetail = JobBuilder
+            .Create(typeof(PubMessageJob))
+            .WithIdentity(jobId)
+            .UsingJobData(GeneralConstants.JOB_QUEUE_KEY, job.QueueName)
+            .StoreDurably()
+            .Build();
+
+        await _scheduler.AddJob(jobDetail, true);
+
+        if (_hostEnvironment.IsDevelopment())
+        {
+            // schedule to run now, once
+            ITrigger jobTrigger = TriggerBuilder
+                .Create()
+                .WithIdentity(Guid.Empty.ToString(), jobId)
+                .WithSimpleSchedule()
+                .ForJob(jobDetail)
+                .Build();
+
+            _ = await _scheduler.ScheduleJob(jobTrigger);
+        }
+        else
+        {
+            foreach (Trigger trigger in job.Triggers)
+            {
+                string triggerId = trigger.Id.ToString();
+
+                if (!CronExpression.IsValidExpression(trigger.CronExpression))
+                {
+                    Trigger faultyTrigger = await orchestratorContext.Triggers
+                        .FirstAsync(x => x.Id == trigger.Id);
+                    faultyTrigger.Status = TriggerStatus.Disabled;
+
+                    await orchestratorContext.SaveChangesAsync();
+                    _logger.LogError(
+                        "Found invalid trigger {TriggerId}. Trigger was disabled",
+                        triggerId);
+
+                    continue;
+                }
+
+                if (trigger.Status == TriggerStatus.Disabled)
+                {
+                    continue;
+                }
+
+                ITrigger jobTrigger = TriggerBuilder
+                    .Create()
+                    .WithIdentity(triggerId, jobId)
+                    .StartNow()
+                    .WithCronSchedule(
+                        trigger.CronExpression,
+                        x =>
+                        {
+                            _ = x
+                                .WithMisfireHandlingInstructionFireAndProceed()
+                                .InTimeZone(TimeZoneInfo.Utc);
+                        })
+                    .ForJob(jobDetail)
+                    .Build();
+
+                _ = await _scheduler.ScheduleJob(jobTrigger);
+
+                _logger.LogInformation(
+                    "Scheduled trigger {TriggerId} with next fire time at {Date}",
+                    triggerId,
+                    jobTrigger.GetNextFireTimeUtc()?.ToString("dd-MM-yyyy HH:ss K"));
```

Comment:
Shouldn't the datetime format have HH:mm instead of HH:ss?

Reply:
It should have been ```HH:mm:ss``` and it is fixed now.